ui/app/models/sync/destination.js

```diff
@@ -11,7 +11,8 @@ export default class SyncDestinationModel extends Model {
   @attr('string', { subText: 'Specifies the name for this destination.' }) name;
   @attr type;
 
-  get icon() {
-    return findDestination(this.type)?.icon;
+  get displayAttributes() {
```

Comment:
if type is undefined the helper `findDestination` returns 

https://github.com/hashicorp/vault/blob/cd2e99a4aa8e94259d212d4d9f09936c22defb1e/ui/lib/core/addon/helpers/sync-destinations.js#L52-L59